### ISSUE TITLE
lora微调qwen3 embedding模型弹出警告find_unused_parameters

### ISSUE THREAD

**Issue by guoxuntao** [modelscope/ms-swift#4698]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)
lora微调qwen3 embedding模型弹出警告find_unused_parameters
使用脚本：
nproc_per_node=8
INFONCE_USE_BATCH=true \
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
INFONCE_MASK_FAKE_NEGATIVE=true \
NPROC_PER_NODE=$nproc_per_node \
swift sft \
    --model ${model_path} \
    --task_type embedding \
    --model_type qwen3_emb \
    --train_type lora \
    --torch_dtype bfloat16 \
    --gradient_checkpointing_kwargs '{"use_reentrant":false}' \
    --dataset  ${data_workspace}/llm4rec_qwen_recall_train.json \
    --split_dataset_ratio 0.0001 \
    --eval_strategy steps \
    --output_dir ${out_workspace} \
    --eval_steps 200 \
    --num_train_epochs 2 \
    --save_steps 200 \
    --per_device_train_batch_size 24 \
    --per_device_eval_batch_size 24 \
    --gradient_accumulation_steps 1 \
    --warmup_ratio 0.05 \
    --logging_steps 200 \
    --learning_rate 2e-4 \
    --loss_type infonce \
    --label_names labels \
    --dataloader_num_workers 32 \
    --dataset_num_proc 32 \
    --dataloader_drop_last true

Train:   0%|                                                                                                                                     | 0/3270 [00:00<?, ?it/s][rank0]:[W624 16:01:26.267456443 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank2]:[W624 16:01:37.864217502 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank4]:[W624 16:01:41.441478311 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank6]:[W624 16:01:45.822980835 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank1]:[W624 16:01:46.006937497 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank5]:[W624 16:01:47.726286279 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank7]:[W624 16:01:47.764196247 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank3]:[W624 16:01:53.269529502 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
cuda 12.6 ubuntu A100 torch 2.6

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by slin000111** [comment]
> **Describe the bug** What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图) lora微调qwen3 embedding模型弹出警告find_unused_parameters 使用脚本： nproc_per_node=8 INFONCE_USE_BATCH=true CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 INFONCE_MASK_FAKE_NEGATIVE=true NPROC_PER_NODE=$nproc_per_node swift sft --model ${model_path} --task_type embedding --model_type qwen3_emb --train_type lora --torch_dtype bfloat16 --gradient_checkpointing_kwargs '{"use_reentrant":false}' --dataset ${data_workspace}/llm4rec_qwen_recall_train.json --split_dataset_ratio 0.0001 --eval_strategy steps --output_dir ${out_workspace} --eval_steps 200 --num_train_epochs 2 --save_steps 200 --per_device_train_batch_size 24 --per_device_eval_batch_size 24 --gradient_accumulation_steps 1 --warmup_ratio 0.05 --logging_steps 200 --learning_rate 2e-4 --loss_type infonce --label_names labels --dataloader_num_workers 32 --dataset_num_proc 32 --dataloader_drop_last true
> 
> Train: 0%| | 0/3270 [00:00<?, ?it/s][rank0]:[W624 16:01:26.267456443 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator()) [rank2]:[W624 16:01:37.864217502 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator()) [rank4]:[W624 16:01:41.441478311 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator()) [rank6]:[W624 16:01:45.822980835 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator()) [rank1]:[W624 16:01:46.006937497 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator()) [rank5]:[W624 16:01:47.726286279 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator()) [rank7]:[W624 16:01:47.764196247 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator()) [rank3]:[W624 16:01:53.269529502 reducer.cpp:1400] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
> 
> **Your hardware and system info** Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等) cuda 12.6 ubuntu A100 torch 2.6
> 
> **Additional context** Add any other context about the problem here(在这里补充其他信息)

没有复现
```shell
nproc_per_node=2
CUDA_VISIBLE_DEVICES=0,1 \
INFONCE_TEMPERATURE=0.1 \
NPROC_PER_NODE=$nproc_per_node \
swift sft \
    --model Qwen/Qwen3-Embedding-0.6B \
    --task_type embedding \
    --model_type qwen3_emb \
    --train_type full \
    --dataset sentence-transformers/stsb:positive \
    --split_dataset_ratio 0.05 \
    --eval_strategy steps \
    --output_dir output \
    --save_steps 50 \
    --eval_steps 50 \
    --num_train_epochs 5 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 4 \
    --learning_rate 6e-6 \
    --loss_type infonce \
    --label_names labels \
    --dataloader_drop_last true \
    --gradient_checkpointing_kwargs '{"use_reentrant":false}'
```
```text
ms_swift                                 3.7.1
transformers                             4.51.3
torch                                    2.6.0
```

**Comment by HuaXiong-Liu** [comment]
Has the problem been resolved?
